sound/soc/sof/hw-spi.c

```diff
@@ -247,6 +247,11 @@ static int spi_sof_probe(struct snd_sof_dev *sdev)
 	unsigned int irq_trigger, irq_sense;
 	int ret;
 
+	if (!spi) {
```

Comment:
I don't think this can really happen. spi could be NULL only if the platform device parent were NULL, and that isn't the case. Was this complained about by klocwork? I'd prefer to add an exception for this than to add redundant checks just to make it happy.

Reply:
we can add an exception but other tools in other locations are going to complain. We might as well fix this and make the tools happy (with an explicit note that the code isn't very useful due to the parent/child model for SOF).